### PR TITLE
fix: decouple playground with risedev

### LINKED ISSUE #7238: bug: failed to run playground mode using docker image
### Describe the bug

failed to `docker run risingwavelabs/risingwave:v0.1.15 playground`, seems RisingWave requires a config file located in `src/config/default.toml`.

### To Reproduce

run 
```bash
 docker run -p 4566:4566 -p 5691:5691 risingwavelabs/risingwave:v0.1.15 playground
```
got
```
launching playground
config file: src/config/default.toml
config file: src/config/default.toml
config file: src/config/default.toml
thread 'main' panicked at 'failed to open config file 'src/config/default.toml': No such file or directory (os error 2)', src/common/src/config.rs:39:29
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

a workaround:
```
docker run --entrypoint "/usr/bin/bash" \
risingwavelabs/risingwave:v0.1.15 \
-c "mkdir -p /src/config && touch /src/config/default.toml && /risingwave/bin/risingwave playground"
```

### Expected behavior

`docker run risingwavelabs/risingwave:v0.1.15 playground` can work without workarounds.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Previously, the `risingwave playground` command trys to load `risedev.yml` to get the profile for playground. It fallbacks to default configurations if `risedev.yml` is not found (Note: users using pre-built binary always go in this path). This can cause confusion because the configuration will be different in different working directory and even bugs like #7238 (Fix #7238)

This PR makes `playground` self-contained, so that users can run the `risingwave` binary with the same configuration in any directory. The code for playground is also more straitforward and easier to change for developers.

Other minor changes about docker:
- Add a check to ensure the image starts correctly
- Remove `docker/build.sh`, which was used to build individual binaries. They are deprecated after #6653
- Fix `docker-compose.yml`: use latest image
- Add some docs about docker

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 

### Release note

- Fix panic for docker users `docker run risingwavelabs/risingwave:v0.1.15 playground`
- Users using pre-built binary can use different playground profiles. (But maybe we don't need to docment this. Just using defaults is fine.)

### CODE REVIEW

**Comment by xxchan** [comment]
- Should we publish a new docker image for v0.1.15? And a new binary v0.1.15 (This seems not very necessary)?
- For https://github.com/risingwavelabs/risingwave/issues/7236, maybe we also hardcode the parameters in code? cc @mikechesterwang

File: src/cmd_all/src/playground.rs [hunk 1]
```diff
@@ -12,36 +12,110 @@
...
+
+/// Embed the config file and create a temporary file at runtime.
+static CONFIG_PATH_WITH_IDLE_EXIT: LazyLock<TempPath> = LazyLock::new(|| {
+    let mut file = tempfile::NamedTempFile::new().expect("failed to create temp config file");
+    write!(
+        file,
+        "[meta]
+disable_recovery = true
+dangerous_max_idle_secs = {IDLE_EXIT_SECONDS}
+max_heartbeat_interval_secs = 600",
```
Comment (BugenZhao) [inline src/cmd_all/src/playground.rs:54]: What about making this a separate file and `include_bytes`?

**Review by BugenZhao — APPROVED** [review]

File: src/cmd_all/src/playground.rs (hunk 1 above)
Comment (xxchan) [reply src/cmd_all/src/playground.rs:54]: What's the benefit? 🤪

Comment (BugenZhao) [reply src/cmd_all/src/playground.rs:54]: Highlighting, and easy to maintain. 😄 We can filter the project with `.toml` files if to modify something.

Comment (BugenZhao) [reply src/cmd_all/src/playground.rs:54]: Oops. You make idle secs a Rust constant. 😰

**Comment by xxchan** [comment]
It seems docker ci is removed in #6692 (Should we also remove `docker.yml`? Or bring back the check? 🤔). cc @huangjw806

**Review by huangjw806 — APPROVED** [review]
LGTM

**Comment by fuyufjh** [comment]
My only concern is that our [Playground](https://playground.risingwave.dev/) seems to be built on the `risingwave playground`. Do they need to tune some parameters? (like listening port, memory, etc.)

**Review by fuyufjh — APPROVED** [review]
LGTM because it at least decouples playground from the source code, so that users can simply download the binary and play with it.

**Pull request merged**